core/src/main/scala/org/apache/spark/storage/BlockManagerMessages.scala

```diff
@@ -108,4 +107,6 @@ private[spark] object BlockManagerMessages {
     extends ToBlockManagerMaster
 
   case class BlockManagerHeartbeat(blockManagerId: BlockManagerId) extends ToBlockManagerMaster
+
+  case class HasCachedBlocks(executorId: String) extends ToBlockManagerMaster
```

Comment:
We still need it for BlockManagerMaster -> BlockManagerMasterEndpoint communication.